internal/report/snapshot/filter_logs.go

```diff
@@ -0,0 +1,122 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshot
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+// filterLogs filters the logs by the given query and timestamp.
+// The returned payload cannot be assumed to be a copy, so it should not be modified.
+func filterLogs(logs plog.Logs, query *string, minTimestamp *time.Time) plog.Logs {
+	// No filters specified, filtered logs are trivially the same as input logs
+	if query == nil && minTimestamp == nil {
+		return logs
```

Comment:
The base case does not return a copy is that a concern?

Reply:
I think not after looking at the rest of the PR as a copy is passed in by I'll keep this question just to make sure.

Reply:
Yeah, the code ends up basically being equivalent if search and mintimestamp are both unspecified.
This function does have a comment above it explaining that a copy cannot be assumed, just in case.